### PR TITLE
Tests for proposal-well-formed-stringify

### PR DESCRIPTION
**Pull request by gibson042**
https://github.com/tc39/proposal-well-formed-stringify

Prevent `JSON.stringify` from returning ill-formed Unicode strings.

Includes moderate refactoring of other `JSON.stringify` tests, which can be separated out upon request.

### CODE REVIEW

**Comment by mathiasbynens** [comment]
The proposal advanced to stage 3 at the September 2018 TC39 meeting. Please remove the “awaiting stage 3” label.

File: test/built-ins/JSON/stringify/15.12.3_4-1-2.js [hunk 1]
```diff
@@ -3,11 +3,15 @@
 
 /*---
 es5id: 15.12.3_4-1-2
-description: JSON.stringify a circular object throws a TypeError
+description: JSON.stringify a indirectly circular object throws a error
```
Comment (mathiasbynens) [inline test/built-ins/JSON/stringify/15.12.3_4-1-2.js]: an

Comment (mathiasbynens) [reply test/built-ins/JSON/stringify/15.12.3_4-1-2.js]: 2×, i.e.

```
JSON.stringify on an indirectly circular object throws an error
```

File: test/built-ins/JSON/stringify/string-escape-ascii.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+esid: sec-quotejsonstring
+description: >
+    JSON.stringify property names and values containing ASCII
+    characters that require escaping
+---*/
+
+var char_to_json = {
+  '"': '\\"',
+  "\\": "\\\\",
+  "\x00": "\\u0000",
```
Comment (mathiasbynens) [inline test/built-ins/JSON/stringify/string-escape-ascii.js]: `\x00` _could_ be `\0` (it’s _not_ an octal escape sequence)

Comment (gibson042) [reply test/built-ins/JSON/stringify/string-escape-ascii.js]: I addressed the grammar nits, but left this `\x00` for internal consistency with the following properties.

**Comment by mathiasbynens** [comment]
This seems good to go. @rwaldron @leobalter Would it be possible to prioritize this PR over others please? V8 is looking to ship this and we'd like to wait until there are official (and merged) Test262 tests.

**Pull request merged**